week17/friday/AngularJS-intro.md

```diff
@@ -0,0 +1,191 @@
+really interact with your data directly at all. Instead, the user interacts 
+with the AngularJS app directly, which then communicates with some other back-end 
+service, which can be an app that the AngularJS is integrated with or an API. 
+
+I find it helpful to think of AngularJS as a “shallow” framework: the AngularJS 
+section of your app should just be a shallow scrape across the top of the app, 
+encompassing your JavaScript, your views, and some routing and controller logic 
+to serve up these views.
+
+
+##AngularJS moves and improves logic
+
+AngularJS moves and improves the logic flow of JavaScript code. JavaScript does 
+a lot of amazing things, but it was created in 10 days by people who did not 
+realize that it would soon become an industry standard and a huge part of web 
+development, so not all of the logic in JavaScript is smooth or useful. There are 
+a lot of strange quirks that JavaScript developers have to learn how to work around.
+
+If you think about regular JavaScript, creating a simple click event is a little 
+messy and hits a lot of different parts of your code. First, it relies on a CSS 
+class as an identifier. Then, you add a watch event to it. In order to add the 
+watch event, you need to look through the DOM to find all instances of the CSS 
+class to add that handler--but before you do that, you need to make sure that the 
+DOM is loaded, which means you have to pause your JavaScript long enough to actually 
+load the DOM.
+
+This might not seem that significant, but one of the really cool parts about 
+JavaScript is that it runs asynchronously. It doesn’t wait around for everything 
+to finish and load, which means it’s really fast and really slick. Pausing your 
+JavaScript functions until your window loads is really slow and clunky.
+
+Once the DOM loads, the event handler is added. Then, when a user clicks on that 
+class, you check and see if there are any attached event handlers. If there are, 
+the appropriate function runs. If you add any more elements to the DOM with this 
+same class after the initial load--like in Lister, where you added a list item, 
+and then added another form once that item was created without refreshing the page, 
+you needed to add a new event handler all over again.
+
+This means that having a single-page application, where the website and the database 
+respond to the actions that you are taking but do not refresh each time, is really
+complicated and takes a lot more effort and maneuvering than a multi-page site.
+
+Your JavaScript view code probably looks like this:
+
+    <div class="button">Say hi!</div>
+
+and the JavaScript page looks like this:
+
+    function init() {
+      $('.button').click(function() {
+        alert('Hi!');
+      });
+    }
+    window.onload = init;
+
+In AngularJS, on the other hand, all of this logic is moved to the view and 
+the controller. If you want to add a click event to a div in AngularJS, you 
```

Comment:
I added a short note about that. We'll cover it more thoroughly next week. Thanks for the feedback!